File: Frogger/fog.cpp
#include "fog.h"

Fog::Fog() {

	index = 0;
	vsml = VSMathLib::getInstance();

	fogParams.isEnabled = false;
	fogParams.fDensity = 0.04f;
	fogParams.vFogColorR = 0.7f;
	fogParams.vFogColorG = 0.7f;
	fogParams.vFogColorB = 0.7f;
	fogParams.fStart = 10.0f;
	fogParams.fEnd = 75.0f;
	fogParams.iEquation = FOG_EQUATION_EXP; // 0 = linear, 1 = exp, 2 = exp2
}

Fog::~Fog() {}

void Fog::draw(VSShaderLib shader) {

	GLint loc;

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.isEnabled");
	glUniform1i(loc, fogParams.isEnabled);

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.vFogColorR");
	glUniform1f(loc, fogParams.vFogColorR);

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.vFogColorG");
	glUniform1f(loc, fogParams.vFogColorG);

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.vFogColorB");
	glUniform1f(loc, fogParams.vFogColorB);

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.fStart");
	glUniform1f(loc, fogParams.fStart);

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.fEnd");
	glUniform1f(loc, fogParams.fEnd);

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.fDensity");
	glUniform1f(loc, fogParams.fDensity);

	loc = glGetUniformLocation(shader.getProgramIndex(), "fogParams.iEquation");
	glUniform1i(loc, fogParams.iEquation);

}

File: FroggerJS/fog.js
function Fog() {
	
	this.fogParams = { isEnabled: false, fDensity: 0.02, vFogColorR: 0.7, vFogColorG: 0.7, vFogColorB: 0.7, fStart: 10.0, fEnd: 75.0, iEquation: 2};
}

Fog.prototype.draw = function() {

		//Enabled bool
		var loc = gl.getUniformLocation(shaderProgram, "fogParams.isEnabled");
		gl.uniform1i(loc, this.fogParams.isEnabled);

		//Fog density
		loc = gl.getUniformLocation(shaderProgram, "fogParams.fDensity");
		gl.uniform1f(loc, this.fogParams.fDensity);

		//Red color
		loc = gl.getUniformLocation(shaderProgram, "fogParams.vFogColorR");
		gl.uniform1f(loc, this.fogParams.vFogColorR);

		//Green color
		loc = gl.getUniformLocation(shaderProgram, "fogParams.vFogColorG");
		gl.uniform1f(loc, this.fogParams.vFogColorG);
		
		//Blue color
		loc = gl.getUniformLocation(shaderProgram, "fogParams.vFogColorB");
		gl.uniform1f(loc, this.fogParams.vFogColorB);
		
		//
		loc = gl.getUniformLocation(shaderProgram, "fogParams.fStart");
		gl.uniform1f(loc, this.fogParams.fStart);
		
		//
		loc = gl.getUniformLocation(shaderProgram, "fogParams.fEnd");
		gl.uniform1f(loc, this.fogParams.fEnd);
		
		//
		loc = gl.getUniformLocation(shaderProgram, "fogParams.iEquation");
		gl.uniform1i(loc, this.fogParams.iEquation);
}

Fog.prototype.switchFog = function() {

	if (this.fogParams.isEnabled) {
		this.fogParams.isEnabled = false;
	}
	else {
		this.fogParams.isEnabled = true;
	}
}